### ISSUE TITLE
Blocking someone unhides their phone number

### ISSUE THREAD

**Issue by hjek** [signalapp/Signal-Android#13824]
Blocking a user will (in some cases) make their phone number visible in the list of blocked users *even if they've chosen that **nobody** should be able to see their phone number.*

Perhaps this is by design, but if it is, then I don't think it's a good design, because when an option screen says

> Your phone number will not be visible to anyone.

then it gives the impression that the phone number will not be visible to others. And the phone number isn't *usually* visible to others, except when the user is blocked, and their phone number is displayed below their name in the list of blocked users.

**Comment by greyson-signal** [comment]
Just to be clear, the only way you could have seen their number is if you already had it. Usually this means that you had contact with them before we launched the phone number privacy settings, or maybe you started the chat with them via phone number, something like that.

That said, we do our best to respect the privacy setting in the UI, even if your device technically knows it. I agree that blocked UI shouldn't show the number, and we made that change in 7.22. 

What version are you running?

**Comment by hjek** [comment]
These contacts have never given me their number, but I may have been in a chat with them before the hidden number setting was available.

That said, some people here believed that the hidden number setting would hide the number after being enabled, and the fact that it wasn't has lead to some very unfortunate situations.

I am glad that it is being fixed. However, will it still be the case that hidden numbers can be made visible *if they were at some point in the past unhidden?*

In that case, I **strongly** suggest rewording the option text because it can provide false sense of security.

I am running 7.26 now. I believe I was running 7.21 at the time.

**Comment by greyson-signal** [comment]
There's various reasons why numbers may be kept under the hood, many of them having to do with linking to contacts, merging users together, etc. 

If a user has phone number sharing enabled, the only time other users will see their number is if they have them in their contacts. That's intentional. Otherwise we absolutely make a point to hide numbers in the UI if a user has phone number sharing off.

If a user wants to avoid all possibilities of having their number known, they can turn off phone number discovery, which will prevent them from being linked to their friend's system contacts.
